Release Notes 17.7 (#24317)
opf/openproject: docs/release-notes/17-7-0/README.md

**Staffing**

The Staffing view allows project managers to assign work while taking each team member's availability and existing workload into account. Resource requests can be allocated to suitable team members, helping organizations distribute work more effectively and avoid overallocations.

![OpenProject's Staffing view shows an open resource request for Laboratory testing together with a matching project member who can be assigned to the work.](openproject-resource-management-staffing-overlay.png)

### Multiple active sprints without sharing (Enterprise add-on)

[feature: multiple_active_sprints]

OpenProject 17.7 introduces support for multiple active sprints within a single project, eliminating the need to share sprints across projects. Teams can now manage multiple active sprints while keeping all sprint planning in one project. This also provides a consolidated overview of active sprints and backlog items on the Backlogs and Sprints page.

### Community improvements for Backlog and Sprints

OpenProject 17.7 also includes several other improvements for Backlog and Sprints that make sprint planning and backlog management more efficient.

> **Add existing work packages to a sprint**

Reviewer: The heading is only speaking of sprints while the paragraph correctly lists the other containers.

You can now add existing work packages directly to sprints, backlog buckets, and the backlog inbox from the Backlogs view. This lets you organize work without leaving the page, helping you stay focused and maintain context while planning your backlog.

**Multi-select for sprint and backlog filters**

> Backlog and Sprint views now support selecting multiple sprints and backlog buckets at once. This makes it easier to focus on the work that matters most by displaying only the relevant sprints and backlog buckets.

Reviewer: Sorry for the nitpicking. But the sentence 'Backlog and Sprint views now support selecting multiple sprints and backlog buckets at once.' should rather be 'Backlog and Sprint views now support selecting multiple sprints and backlog buckets to show'. The 'at once' does not convey what is possible.

Author: @ulferts Nitpicking is much appreciated in this case, thank you. For this release, I have even less time to prepare and can't really test all features thoroughly, so every review is very helpful. 🙏

**Improved drag and drop**

> Drag-and-drop interactions in the Backlogs and Sprints view have been improved to provide a smoother planning experience across desktop and mobile devices. Moving work packages between the backlog and sprints, changing their order, or reorganizing the hierarchy is now more intuitive and reliable.

Reviewer: Changing the hierarchy is not possible on backlogs.

The timeline view visualizes each user's workload over time, helping project managers understand resource utilization, identify conflicts early, and make informed planning decisions. Project managers can **allocate resources directly from the timeline**, streamlining resource planning and staffing.

![OpenProject's Resource planner timeline visualizes allocated work and remaining capacity for each user over time, helping identify overallocations and available resources.](openproject-resource-management-timeline-allocated.png)

**Staffing**

The Staffing view allows project managers to assign work while taking each team member's availability and existing workload into account. Resource requests can be matched with suitable team members, helping organizations distribute work more effectively and avoid overallocations.

![OpenProject's Staffing view shows an open resource request for Laboratory testing together with a matching project member who can be assigned to the work.](openproject-resource-management-staffing-overlay.png)

### Multiple active sprints without sharing (Enterprise add-on)

[feature: multiple_active_sprints]

> OpenProject 17.7 introduces support for multiple active sprints without requiring work packages to be shared between projects. Teams can now run multiple active sprints independently while keeping work packages within their respective projects. This provides greater flexibility for organizations managing multiple Scrum teams or parallel development efforts.

Reviewer: The objects that are shared in the context of backlogs are the sprints, never the work packages. 

The value of this feature for users is that they can have multiple sprints active in a single project. They don't need to set up a multi project hierarchy and share the sprints. Having all in one project also means that the backlogs and sprints page shows everything that is going on. This would not be possible with sprint sharing as the backlogs and sprints page is currently limited to displaying the current project's sprints and work packages. Of course, the full picture could still be seen using the work package table. But multiple active sprints makes sense if you want to keep everything in the same project.

### Community improvements for Backlog and Sprints

> OpenProject 17.7 also includes several community-contributed improvements for Backlog and Sprints that make sprint planning and backlog management more efficient.

Reviewer: The features in backlogs are not community contributed.

Author: Thank you, I thought I updated that already.

**Add existing work packages to a sprint**

> You can now add existing work packages to a sprint directly from the Backlog view. This makes it easier to move planned work into a sprint without changing the work package hierarchy or creating duplicate work packages.

Reviewer: Existing and new work packages can be added to all containers: sprints, backlog buckets and the backlog inbox. The value is not so much to avoid changes in hierarchy or duplication but rather that it enables the user to add work packages without leaving the page and by that losing the context.

**Multi-select filters**

> Backlog and Sprint filters now support selecting multiple values for the same filter criterion. This makes it easier to filter work packages by multiple assignees, priorities, statuses, versions, or other attributes.

Reviewer: I am not sure if this feature is meant: [Add multi-select drop-down for sprint and backlog buckets](https://community.openproject.org/projects/AGILE/work_packages/AGILE-176/activity)

If it is, it is about filtering which sprints and backlog buckets to display. 

<img width="1604" height="636" alt="Image" src="https://github.com/user-attachments/assets/6882cf8c-1a05-4a67-8e06-352aed33bda7" />

But there are no filters beyond that (as of now - working on it). The work packages of the selected sprints and backlogs will always be shown in full. 

The user value of this one being able to focus on the currently important sprint/backlog instead of having to ignore what is currently not interresting.

**Improved drag and drop**

> Drag-and-drop interactions in the Backlog and Sprint views have been improved to provide a smoother planning experience. Moving work packages between the backlog and sprints, changing their order, or reorganizing the hierarchy is now more intuitive and reliable.

Reviewer: It also works on mobile incl tablet.

### Improvements for PM² and PMflex management

OpenProject 17.7 introduces several improvements for organizations using the PM² and PMflex project management methodologies, making it easier to manage project information and monitor project progress.

**Show project attributes as separate tab in a work package**

Project attributes can now be displayed in a dedicated **Project attributes** tab within work packages. This provides quicker access to project-specific information while keeping work package details and project metadata clearly separated.

![OpenProject displays project attributes in a dedicated Project attributes tab within a Project initiation request work package.](openproject-17.7-project-attributes-tab.png)

**PMflex Artefact PDF export**

Work packages can now be exported using a dedicated **PMflex Artefact** PDF template. The export combines project attributes, work package attributes, custom fields, and related work packages into a structured document suitable for project documentation and governance.

> Administrators can configure PMflex artefacts to be generated automatically when a work package reaches a defined status. Generated artefacts can either be uploaded to the project's connected Nextcloud folder or added directly as work package attachments.

Reviewer 2: It does not have to be Nextcloud but could also be Onedrive, so basically whatever storage provider was chosen.

Author: @HDinger That's interesting because in [this work package](https://community.openproject.org/wp/DREAM-763), "Providers other than Nextcloud (OneDrive/SharePoint uploads)" is listed under "out of scope"?

Reviewer 2: Mhh.. good point. I was not aware of that. @as-op can probably tell best what is actually implemented.

Reviewer 2: > Mhh.. good point. I was not aware of that. @as-op can probably tell best what is actually implemented.

Ah, he already wrote in the ticket that it only works for Nextcloud. Then forgot what I said 😅

Reviewer 3: I don't know the reason why, someone else implemented the uploading to external storage while the hackathon. 
This limitation is in the PIR export. As I'm using the same code to upload the new artefact to the project folder, this also applies for it.
<img width="724" height="380" alt="Bildschirmfoto 2026-07-20 um 13 58 47" src="https://github.com/user-attachments/assets/2ec14430-ab7c-473c-96b1-7c488af5c825" />

**Project phases and gates widget**

A new Project phases and gates widget is available for project overviews. It provides a visual representation of the project's current phase and gate status, helping project managers and stakeholders quickly understand project progress at a glance.

![OpenProject's Project phases and gates widget visualizes the current project phase and gate status on the project overview page.](openproject-17-7-phases-gates-widget.png)

### Work package previews from URLs in Documents

> When working with the BlockNote editor in Documents, links to work packages now display rich previews instead of plain URLs. This makes it easier to identify referenced work packages and understand their context without leaving the document.

Reviewer 4: Hmm, no that is not the actual feature. We already had work package links instead of plain URLs. We already were able to add a link from a document to work packages. 

What is new:

- You can _paste_ a work package URL into a document and it will be converted into a nicely rendered work package link.
- Work package links, can be rendered in various sizes. Those that are rendered as "tiny", only show the work package ID, which often is not enough information. This release now provides a preview on mouse hover, revealing information of the linked work package, such as the subject and type.

> ### Semantic identifiers now support BCF import and export

Reviewer 4: ```suggestion
### The BCF import and export now support the semantic identifier (ABC-123)
```

> Semantic identifiers are now considered production-ready and are no longer marked as Beta. OpenProject 17.7 also adds support for importing and exporting semantic identifiers in BCF files.

Reviewer 4: ```suggestion
Semantic identifiers are now considered production-ready and are no longer marked as Beta. OpenProject 17.7 also adds support for importing and exporting BCF files having semantic identifiers.
```

### Filter projects by portfolio and program (Enterprise add-on)

[feature: portfolio_management]

Project lists now support filtering by portfolio and program. This makes it easier to find projects within large project portfolios and provides greater flexibility when creating project overviews and reports.

### Additional calculation operators for calculated fields

> Calculated custom fields now support additional calculation operators, allowing you to create more advanced formulas and model a wider range of business logic directly in OpenProject.

Reviewer: It also supports formulas and keywords now. Not sure if it is worth mentioning but maybe it would be good to just call it 'calculation options' or 'calculation capabilities'  and provide examples like power, average and sum. Those are capabilities people might be looking for as they are used to it from xls.

<!-- END AUTOMATED SECTION -->
<!-- Warning: Anything above this line will be automatically removed by the release script -->

## Contributions

A very special thank you goes to Helmholtz-Zentrum Berlin, City of Cologne, Deutsche Bahn, ZenDiS, and STEF for sponsoring released or upcoming features. Your support, alongside the efforts of our amazing Community, helps drive these innovations.

> Also a big thanks to our Community members for reporting bugs and helping us identify and provide fixes. Special thanks for reporting and finding bugs go to Rince wind, Walid Ibrahim, Gábor Alexovics, Brandon Soonaye, and Mohammed Mohiuddin.

Reviewer 2: ```suggestion
Also a big thanks to our Community members for reporting bugs and helping us identify and provide fixes. Special thanks for reporting and finding bugs go to Rince Wind, Walid Ibrahim, Gábor Alexovics, Brandon Soonaye, and Mohammed Mohiuddin.
```
